Description: Remove overreaching citation notice
 Quoting the gpl-faq:
 .
 > Does the GPL allow me to add terms that would require citation or
 > acknowledgment in research papers which use the GPL-covered software or its
 > output? (#RequireCitation)
 > .
 > No, this is not permitted under the terms of the GPL. While we recognize
 > that proper citation is an important part of academic publications,
 > citation cannot be added as an additional requirement to the GPL. Requiring
 > citation in research papers which made use of GPL'd software goes beyond
 > what would be an acceptable additional requirement under section 7(b) of
 > GPLv3, and therefore would be considered an additional restriction under
 > Section 7 of the GPL. And copyright law does not allow you to place such a
 > requirement on the output of software, regardless of whether it is licensed
 > under the terms of the GPL or some other license.
 .
 Therefore, removing this to make parallel GPL-compliant.

 While I agree with the GPL FAQ I found the following change inside the
 code while refreshing the patch:
 .
     # Before changing this line, please read
     # https://www.gnu.org/software/parallel/parallel_design.html#Citation-notice
     # https://git.savannah.gnu.org/cgit/parallel.git/tree/doc/citation-notice-faq.txt
 .
 Thus this patch is again up for discussion.

Author: Didier Raboud <odyx@debian.org>
        Ian Turner <vectro@vectro.org> (for the documentation parts)
Bug-Debian: https://bugs.debian.org/905674

Last-Update: 2021-08-28

--- a/src/env_parallel.dash
+++ b/src/env_parallel.dash
@@ -395,7 +395,7 @@ _parset_main() {
 	echo "Web site: https://www.gnu.org/software/parallel"
 	echo
 	echo "When using programs that use GNU Parallel to process data for publication"
-	echo "please cite as described in 'parallel --citation'."
+	echo "please cite as described in the manpage."
 	echo
 	return 255
     fi
--- a/src/env_parallel.ksh
+++ b/src/env_parallel.ksh
@@ -373,7 +373,7 @@ _parset_main() {
 	echo "Web site: https://www.gnu.org/software/parallel"
 	echo
 	echo "When using programs that use GNU Parallel to process data for publication"
-	echo "please cite as described in 'parallel --citation'."
+	echo "please cite as described in the manpage."
 	echo
 	return 255
     fi
--- a/src/env_parallel.pod
+++ b/src/env_parallel.pod
@@ -800,9 +800,6 @@ When using GNU B<env_parallel> for a pub
 O. Tange (2018): GNU Parallel 2018, March 2018, ISBN 9781387509881,
 DOI: 10.5281/zenodo.1146014.
 
-This helps funding further development; and it won't cost you a cent.
-If you pay 10000 EUR you should feel free to use GNU Parallel without citing.
-
 Copyright (C) 2007-10-18 Ole Tange, http://ole.tange.dk
 
 Copyright (C) 2008-2010 Ole Tange, http://ole.tange.dk
--- a/src/env_parallel.sh
+++ b/src/env_parallel.sh
@@ -400,7 +400,7 @@ _parset_main() {
 	echo "Web site: https://www.gnu.org/software/parallel"
 	echo
 	echo "When using programs that use GNU Parallel to process data for publication"
-	echo "please cite as described in 'parallel --citation'."
+	echo "please cite as described in the manpage."
 	echo
 	return 255
     fi
--- a/src/env_parallel.zsh
+++ b/src/env_parallel.zsh
@@ -365,7 +365,7 @@ _parset_main() {
 	echo "Web site: https://www.gnu.org/software/parallel"
 	echo
 	echo "When using programs that use GNU Parallel to process data for publication"
-	echo "please cite as described in 'parallel --citation'."
+	echo "please cite as described in the manpage."
 	echo
 	return 255
     fi
--- a/src/parallel
+++ b/src/parallel
@@ -1607,7 +1607,7 @@ sub options_hash() {
 	 # Before changing this line, please read
 	 # https://www.gnu.org/software/parallel/parallel_design.html#Citation-notice
 	 # https://git.savannah.gnu.org/cgit/parallel.git/tree/doc/citation-notice-faq.txt
-	 "bibtex|citation" => \$opt::citation,
+#	 "bibtex|citation" => \$opt::citation,
 	 "wc|willcite|will-cite|nn|nonotice|no-notice" => \$opt::willcite,
 	 # Termination and retries
 	 "halt-on-error|halt=s" => \$opt::halt,
@@ -1764,10 +1764,10 @@ sub parse_options(@) {
     # Before changing this line, please read
     # https://www.gnu.org/software/parallel/parallel_design.html#Citation-notice
     # https://git.savannah.gnu.org/cgit/parallel.git/tree/doc/citation-notice-faq.txt
-    if(defined $opt::citation) {
-	citation(\@argv_before,\@ARGV);
-	wait_and_exit(0);
-    }
+#    if(defined $opt::citation) {
+#	citation(\@argv_before,\@ARGV);
+#	wait_and_exit(0);
+#    }
     # no-* overrides *
     if($opt::nokeeporder) { $opt::keeporder = undef; }
 
@@ -2117,7 +2117,7 @@ sub parse_options(@) {
     #
     # If you want GNU Parallel to be maintained in the future you
     # should keep this line.
-    citation_notice();
+#    citation_notice();
     # Seriously: _YOU_ will be harming free software by removing the
     # notice.  _YOU_ make it harder to justify spending time developing
     # it. If you *do* remove the line, please email
@@ -5058,9 +5058,9 @@ sub usage() {
 	 # Before changing this line,  please read
          # https://www.gnu.org/software/parallel/parallel_design.html#Citation-notice
 	 # https://git.savannah.gnu.org/cgit/parallel.git/tree/doc/citation-notice-faq.txt
-	 "This helps funding further development; AND IT WON'T COST YOU A CENT.",
-	 "If you pay 10000 EUR you should feel free to use GNU Parallel without citing.",
-	 "",
+#	 "This helps funding further development; AND IT WON'T COST YOU A CENT.",
+#	 "If you pay 10000 EUR you should feel free to use GNU Parallel without citing.",
+#	 "",
 	 "",);
 }
 
@@ -5187,7 +5187,7 @@ sub version() {
 	 "",
 	 "Web site: https://www.gnu.org/software/${Global::progname}\n",
 	 "When using programs that use GNU Parallel to process data for publication",
-	 "please cite as described in 'parallel --citation'.\n",
+	 "please cite as described in the manpage.\n",
         );
 }
 
--- a/src/parallel.pod
+++ b/src/parallel.pod
@@ -576,33 +576,6 @@ See also: B<--fg>, B<man sem>.
 
 Implies B<--semaphore>.
 
-
-=item B<--bibtex>
-
-=item B<--citation>
-
-Print the citation notice and BibTeX entry for GNU B<parallel>,
-silence citation notice for all future runs, and exit. It will not run
-any commands.
-
-If it is impossible for you to run B<--citation> you can instead use
-B<--will-cite>, which will run commands, but which will only silence
-the citation notice for this single run.
-
-If you use B<--will-cite> in scripts to be run by others you are
-making it harder for others to see the citation notice.  The
-development of GNU B<parallel> is indirectly financed through
-citations, so if your users do not know they should cite then you are
-making it harder to finance development. However, if you pay 10000
-EUR, you have done your part to finance future development and should
-feel free to use B<--will-cite> in scripts.
-
-If you do not want to help financing future development by letting
-other users see the citation notice or by paying, then please use
-another tool instead of GNU B<parallel>. You can find some of the
-alternatives in B<man parallel_alternatives>.
-
-
 =item B<--block> I<size>
 
 =item B<--block-size> I<size>
@@ -5562,9 +5535,6 @@ When using GNU B<parallel> for a publica
 O. Tange (2011): GNU Parallel - The Command-Line Power Tool, ;login:
 The USENIX Magazine, February 2011:42-47.
 
-This helps funding further development; and it won't cost you a cent.
-If you pay 10000 EUR you should feel free to use GNU Parallel without citing.
-
 Copyright (C) 2007-10-18 Ole Tange, http://ole.tange.dk
 
 Copyright (C) 2008-2010 Ole Tange, http://ole.tange.dk
--- a/src/parallel_alternatives.pod
+++ b/src/parallel_alternatives.pod
@@ -3613,9 +3613,6 @@ When using GNU B<parallel> for a publica
 O. Tange (2011): GNU Parallel - The Command-Line Power Tool, ;login:
 The USENIX Magazine, February 2011:42-47.
 
-This helps funding further development; and it won't cost you a cent.
-If you pay 10000 EUR you should feel free to use GNU Parallel without citing.
-
 Copyright (C) 2007-10-18 Ole Tange, http://ole.tange.dk
 
 Copyright (C) 2008-2010 Ole Tange, http://ole.tange.dk
--- a/src/parallel_design.pod
+++ b/src/parallel_design.pod
@@ -1340,18 +1340,6 @@ and if it works, you should post it on t
 cost work and which have not been tested are, however, unlikely to be
 prioritized.
 
-Running B<parallel --citation> one single time takes less than 10
-seconds, and will silence the citation notice for future runs. This is
-comparable to graphical tools where you have to click a checkbox
-saying "Do not show this again". But if that is too much trouble for
-you, why not use one of the alternatives instead?  See a list in:
-B<man parallel_alternatives>.
-
-As the request for citation is not a legal requirement this is
-acceptable under GPLv3 and cleared with Richard M. Stallman
-himself. Thus it does not fall under this:
-https://www.gnu.org/licenses/gpl-faq.en.html#RequireCitation
-
 
 =head1 Ideas for new design
 
--- a/src/parallel_tutorial.pod
+++ b/src/parallel_tutorial.pod
@@ -2993,9 +2993,6 @@ Output:
   
     O. Tange (2011): GNU Parallel - The Command-Line Power Tool,
     ;login: The USENIX Magazine, February 2011:42-47.
-  
-  This helps funding further development; AND IT WON'T COST YOU A CENT.
-  If you pay 10000 EUR you should feel free to use GNU Parallel without citing.
 
 When asking for help, always report the full output of this:
 
@@ -3013,7 +3010,7 @@ Output:
   Web site: https://www.gnu.org/software/parallel
   
   When using programs that use GNU Parallel to process data for publication
-  please cite as described in 'parallel --citation'.
+  please cite as described in the manpage.
 
 In scripts B<--minversion> can be used to ensure the user has at least
 this version:
@@ -3026,38 +3023,6 @@ Output:
   20160322
   Your version is at least 20130722.
 
-If you are using GNU B<parallel> for research the BibTeX citation can be
-generated using B<--citation>:
-
-  parallel --citation
-
-Output:
-
-  Academic tradition requires you to cite works you base your article on.
-  When using programs that use GNU Parallel to process data for publication
-  please cite:
-  
-  @article{Tange2011a,
-    title = {GNU Parallel - The Command-Line Power Tool},
-    author = {O. Tange},
-    address = {Frederiksberg, Denmark},
-    journal = {;login: The USENIX Magazine},
-    month = {Feb},
-    number = {1},
-    volume = {36},
-    url = {https://www.gnu.org/s/parallel},
-    year = {2011},
-    pages = {42-47},
-    doi = {10.5281/zenodo.16303}
-  }
-  
-  (Feel free to use \nocite{Tange2011a})
-  
-  This helps funding further development; AND IT WON'T COST YOU A CENT.
-  If you pay 10000 EUR you should feel free to use GNU Parallel without citing.
-  
-  If you send a copy of your published article to tange@gnu.org, it will be
-  mentioned in the release notes of next version of GNU Parallel.
   
 With B<--max-line-length-allowed> GNU B<parallel> will report the maximal
 size of the command line:
@@ -3150,7 +3115,7 @@ If you use GNU B<parallel> for research:
 
 =item *
 
-Please cite GNU B<parallel> in you publications (use B<--citation>)
+Please cite GNU B<parallel> in you publications
 
 =back
 
--- a/src/parset.pod
+++ b/src/parset.pod
@@ -195,9 +195,6 @@ When using GNU B<parallel> for a publica
 O. Tange (2011): GNU Parallel - The Command-Line Power Tool, ;login:
 The USENIX Magazine, February 2011:42-47.
 
-This helps funding further development; and it won't cost you a cent.
-If you pay 10000 EUR you should feel free to use GNU Parallel without citing.
-
 Copyright (C) 2007-10-18 Ole Tange, http://ole.tange.dk
 
 Copyright (C) 2008-2010 Ole Tange, http://ole.tange.dk
--- a/src/parsort
+++ b/src/parsort
@@ -402,7 +402,7 @@ sub version() {
 	 "",
 	 "Web site: https://www.gnu.org/software/${Global::progname}\n",
 	 "When using programs that use GNU Parallel to process data for publication",
-	 "please cite as described in 'parallel --citation'.\n",
+	 "please cite as described in the manpage.\n",
         );
 }
 
--- a/src/sem
+++ b/src/sem
@@ -1607,7 +1607,7 @@ sub options_hash() {
 	 # Before changing this line, please read
 	 # https://www.gnu.org/software/parallel/parallel_design.html#Citation-notice
 	 # https://git.savannah.gnu.org/cgit/parallel.git/tree/doc/citation-notice-faq.txt
-	 "bibtex|citation" => \$opt::citation,
+	 # "bibtex|citation" => \$opt::citation,
 	 "wc|willcite|will-cite|nn|nonotice|no-notice" => \$opt::willcite,
 	 # Termination and retries
 	 "halt-on-error|halt=s" => \$opt::halt,
@@ -1764,10 +1764,10 @@ sub parse_options(@) {
     # Before changing this line, please read
     # https://www.gnu.org/software/parallel/parallel_design.html#Citation-notice
     # https://git.savannah.gnu.org/cgit/parallel.git/tree/doc/citation-notice-faq.txt
-    if(defined $opt::citation) {
-	citation(\@argv_before,\@ARGV);
-	wait_and_exit(0);
-    }
+    # if(defined $opt::citation) {
+    #     citation(\@argv_before,\@ARGV);
+    #     wait_and_exit(0);
+    # }
     # no-* overrides *
     if($opt::nokeeporder) { $opt::keeporder = undef; }
 
@@ -2117,7 +2117,7 @@ sub parse_options(@) {
     #
     # If you want GNU Parallel to be maintained in the future you
     # should keep this line.
-    citation_notice();
+    # citation_notice();
     # Seriously: _YOU_ will be harming free software by removing the
     # notice.  _YOU_ make it harder to justify spending time developing
     # it. If you *do* remove the line, please email
@@ -5058,8 +5058,8 @@ sub usage() {
 	 # Before changing this line,  please read
          # https://www.gnu.org/software/parallel/parallel_design.html#Citation-notice
 	 # https://git.savannah.gnu.org/cgit/parallel.git/tree/doc/citation-notice-faq.txt
-	 "This helps funding further development; AND IT WON'T COST YOU A CENT.",
-	 "If you pay 10000 EUR you should feel free to use GNU Parallel without citing.",
+	 # "This helps funding further development; AND IT WON'T COST YOU A CENT.",
+	 # "If you pay 10000 EUR you should feel free to use GNU Parallel without citing.",
 	 "",
 	 "",);
 }
@@ -5187,7 +5187,7 @@ sub version() {
 	 "",
 	 "Web site: https://www.gnu.org/software/${Global::progname}\n",
 	 "When using programs that use GNU Parallel to process data for publication",
-	 "please cite as described in 'parallel --citation'.\n",
+	 "please cite as described in the manpage.\n",
         );
 }
 
